### PR TITLE
INTEXT-158: JavaConfig based UT Cases are added

### PR DESCRIPTION
**Pull request by erenavsarogullari**
JIRA : https://jira.spring.io/browse/INTEXT-158

JavaConfig based Unit Test Cases are added to cover all inbound & outbound adapters.

### CODE REVIEW

**Comment by artembilan** [comment]
Hello, @erenavsarogullari !

Welcome back and thank you for the contribution!

Haven't heard from your side for a while already, but I hope everything is fine with you.

Best wishes and reaching a new tops this year!

Will review your OR soon.

File: spring-integration-hazelcast/README.md [hunk 1]
```diff
@@ -150,12 +152,46 @@ Sample definitions are as follows :
...
 ```
-**Reference :** http://docs.hazelcast.org/docs/3.4/manual/html-single/hazelcast-documentation.html#distributed-data-structures 
+
+#### JavaConfig Driven Configuration :
+
+The following sample shows Distributed Map configuration. Same configuration can be used for other distributed data structures(IMap, MultiMap, ReplicatedMap, IList, ISet, IQueue and ITopic).
+```
+@Bean
+public PollableChannel distributedMapChannel() {
+	return new QueueChannel();
```
Comment (artembilan) [inline spring-integration-hazelcast/README.md]: Indents issue everywhere here in doc for the code samples.

File: spring-integration-hazelcast/build.gradle [hunk 2]
```diff
@@ -41,7 +41,7 @@ ext {
 	hazelcastVersion = '3.5'
```
Comment (artembilan) [inline spring-integration-hazelcast/build.gradle]: I see the fresh Hazelcast 3.6 version!

Plus we can upgrade to Gradle 2.5, but yes, in the separate commit.
Or I can address that on merge.

File: spring-integration-hazelcast/src/test/java/org/springframework/integration/hazelcast/outbound/util/HazelcastOutboundChannelAdapterTestUtils.java [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    public static final String TEST_SURNAME = "Test_Surname";
+
+    public static void testWriteToDistributedMap(MessageChannel channel,
+        Map<?, ?> distributedMap,
+        HazelcastTestRequestHandlerAdvice requestHandlerAdvice) {
+        try {
+            sendMessageToChannel(channel);
+            assertTrue(requestHandlerAdvice.executeLatch.await(10, TimeUnit.SECONDS));
+            verifyMapForPayload(new TreeMap(distributedMap));
+        } catch (InterruptedException e) {
```
Comment (artembilan) [inline spring-integration-hazelcast/src/test/java/org/springframework/integration/hazelcast/outbound/util/HazelcastOutboundChannelAdapterTestUtils.java]: `else`, `catch`, `finally` from new line.
Minor I can address on merge.

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-integration-extensions/commit/0efd494337d0b3a45b79a6cd65da79e585042e27.

@erenavsarogullari , thank you very much for the contribution one more time!

Feel free to come back to us with any other ideas in your mind to go ahead here and everywhere.

**Comment by erenavsarogullari** [comment]
Many thanks @artembilan .

I was fine but too busy due to my personal situations.

Thank you very much for all your wishes ;) Wishing you good health, happiness, and success in the this year and always ;)

Sure, will keep in touch asap ;)

**Pull request closed without merging**